contracts/EthProxy.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.6.7;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "./IUSM.sol";
+import "./external/IWETH9.sol";
+
+
+contract EthProxy {
```

Comment:
And how about here?  `contract EthProxy is IUSM`?

Reply:
Not here. `EthProxy` is a standalone contract, that contains a `IUSM` link inside. Same as `USM` doesn't inherit from `WETH9`, it takes its address as a constructor parameter and then calls it when needed.

It's important to understand that `EthProxy` is completely independent from `USM` and has no special access beyond what each individual user allows.

Reply:
Interesting.  I was operating on a pre-smart-contract intuition where one might define an interface `IUSM` and then have two implementations: one that takes WETH, and another that takes ETH (and passes the input to a wrapped instance of the one that takes WETH).  And then other functions/code could operate on variables of type `IUSM`, without caring which of the two concrete types the var's value was an instance of...

...But after reading a bit more, this seems ambitious for Solidity's/the EVM's type system, and probably overkill for our purposes anyway.  Suggestion retracted!